types/coin.go

```diff
@@ -632,6 +634,23 @@ func ParseCoins(coinsStr string) (coins Coins, err error) {
 	return coins, nil
 }
 
+func IsCoinNameValid(coinName string) bool  {
+	if !reCoinNameCompiled.MatchString(coinName) {
+		return false
+	}
+	return true
+}
+
+func IsCoinMinDenomValid(denom string) bool  {
+	if denom != IrisAtto && !strings.HasSuffix(denom, MinDenomSuffix) {
+		return false
+	}
```

Comment:
Remaining lines can be simplified as:
`return reDenomCompiled.MatchString(denom)`